app/src/main/java/com/alphawallet/app/viewmodel/WalletViewModel.java

```diff
@@ -315,15 +312,21 @@ private void onFetchTokensCompletable()
      */
     private void updateTokenBalances()
     {
-        unknownAddresses.addAll(ethereumNetworkRepository.getAllKnownContracts(tokensService.getNetworkFilters()));
-        checkUnknownAddresses();
+        AddUnresolvedContracts();
         if (balanceTimerDisposable == null || balanceTimerDisposable.isDisposed())
         {
             balanceTimerDisposable = Observable.interval(0, BALANCE_CHECK_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
                     .doOnNext(l -> checkBalances()).subscribe();
         }
     }
 
+    private void AddUnresolvedContracts()
```

Comment:
Matter of style. Start `AddUnresolvedContracts()` with a lowercase instead?

Reply:
Yes probably.